github_helper.py

```diff
@@ -0,0 +1,159 @@
+"""
+github_helper provides several methods to help interface with Github
+"""
+
+import requests
+import urlparse
+
+GITHUB_ORG = 'apache'
+SECRET_FILE = '../../github_auth/apache-beam.secret'
+SOURCE_REMOTE = 'github'
+TARGET_BRANCH = 'master'
+TARGET_REMOTE = 'apache'
+
+GITHUB_API_ROOT = 'https://api.github.com'
+GITHUB_REPO_FMT_URL = GITHUB_API_ROOT + '/repos/{0}/{1}'
+GITHUB_COMMENTS_ENDPOINT = '/{}/comments'
+GITHUB_ISSUES_ENDPOINT = '/issues/'
+GITHUB_PULLS_ENDPOINT = '/pulls/'
+
+BOT_NAME = 'merge-bot'
+GITHUB_SECRET = '../../github_auth/mergebot.secret'
+
+
+class GithubHelper(object):
+    """
+    GithubHelper provides a lightweight authenticated wrapper for the Github
+    API.
+    """
+
+    def __init__(self, org, repo):
+        self.org = org
+        self.repo = repo
+        with open(GITHUB_SECRET, 'r') as key_file:
+            self.bot_key = key_file.read().strip()
+        print('Loaded key file.')
+
+    def fetch_prs(self):
+        """ Gets the PRs for the configured repository
+
+        Returns:
+            An iterable containing instances of GithubPR if successful.
+            None otherwise.
+        """
+        prs = self.get(GITHUB_PULLS_ENDPOINT)
+        if prs is None:
+            return None
+        return [GithubPR(self, pr) for pr in prs]
+
+    def get(self, endpoint):
+        """
+        get makes a GET request against a specified endpoint.
+
+        Args:
+            endpoint: URL to which to make the GET request. URL is relative to
+            https://api.github.com/repos/{self.org}/{self.repo}
+        Returns:
+            JSON object retrieved from the endpoint.
+        """
+        url = urlparse.urljoin(self._repo_url(), endpoint)
+        print('Loading from Github at {}.'.format(url))
+        req = requests.get(url, auth=(BOT_NAME, self.bot_key))
+        if req.status_code != 200:
+            print('Oops, that didn\'t work. Error below, moving on.')
+            print(req.text)
+            return None
```

Comment:
Should this throw an exception instead? Also, the error behavior should be documented in the `Returns` or `Throws` blocks.